src/rgw/rgw_common.cc

```diff
@@ -1309,8 +1309,8 @@ int RGWUserCaps::add_from_string(const string& str)
 {
   int start = 0;
   do {
-    int end = str.find(';', start);
-    if (end < 0)
+    auto end = str.find(';', start);
+    if (end == string::npos)
```

Comment:
I'm pretty sure you can just pass std::string::npos to std::string::substr() and it will do what you want. Note that here "end" is going to be "one past length" once you assign it to size(); since there's no NULL terminator, it will be like npos anyway.